pennylane/optimize/lie_gradient.py

```diff
@@ -0,0 +1,352 @@
+        objective function value prior to the step.
+
+        Args:
+            *args:
+            **kwargs:
+
+        Returns:
+
+        """
+        omegas = self.get_omegas()
+        non_zero_lie_algebra_elements = []
+        non_zero_omegas = []
+        for i, element in enumerate(omegas):
+            if not np.isclose(element, 0):
+                non_zero_lie_algebra_elements.append(
+                    tuple(
+                        self.pauli_str_to_int[p]
+                        for p in self.lie_algebra_basis_names[i]
+                    )
+                )
+                non_zero_omegas.append(-omegas[i])
+        lie_gradient = qml.Hamiltonian(
+            non_zero_omegas,
+            [self.get_pauli_op(ps) for ps in non_zero_lie_algebra_elements],
+        )
+        temp_circuit = copy.copy(self.circuit)
+
+        @append_time_evolution(lie_gradient, self.stepsize)
+        def circuit(params, wires, **kwargs):
+            """Append the time evolution"""
+            for obj in temp_circuit.qtape.operations:
+                qml.apply(obj)
+
+            return qml.state()
+
+        # TODO: qml.map is not working here, it throws an error for pauli @ pauli objects, which have `name` tensor.
+        # cost_fn = qml.map(circuit, self.observables, device=self.circuit.device, measure='expval')([],[])
+        # cost_fn = qml.QNodeCollection(circuit, self.observables, device=self.circuit.device, measure='expval')([],[])
+        # cost = sum([self.coeffs[i] * obs for i, obs in enumerate(cost_fn)])
+        # print(cost)
+
+        self.circuit = qml.QNode(circuit, temp_circuit.device)
+        return self.circuit([], [])
+
+    def step(
+        self,
+        *args,
+        **kwargs,
+    ):
+        r"""Update the circuit with one step of the optimizer
+
+        Args:
+            *args:
+            **kwargs:
+
+        Returns:
+
+        """
+        self.step_and_cost(
+            *args,
+            **kwargs,
+        )
+
+        return None
+
+    def get_su_n_operators(self):
+        r"""Get the 2x2 SU(N) operators. The dimension of the group is N^2-1.
+
+        Returns:
+            List of (N^2)x(N^2) numpy complex arrays and corresponding paulis words
+        """
+
+        operators = []
+        names = []
+        # get all possible combinations of paulis
+        pauli_combinations = it.product(
+            list(range(len(self.pl_paulis))), repeat=self.nqubits
+        )
+        # remove the identity
+        next(pauli_combinations)
+        # construct the corresponding pennylane observables
+        for ps in pauli_combinations:
+            operators.append(self.get_pauli_op(ps))
+            names.append(ps)
+        return operators, [
+            "".join([self.pauli_int_to_str[i] for i in n]) for n in names
+        ]
+
+    def get_pauli_op(self, paulistring):
+        r"""Create a qml.Observable for a String corresponding to a Pauli word.
```

Comment:
likewise, could this be replaced with [`string_to_pauli_word`](https://pennylane.readthedocs.io/en/latest/code/api/pennylane.grouping.string_to_pauli_word.html)?